enterprise/micronaut/src/org/netbeans/modules/micronaut/db/MicronautEndpointTestGenerator.java

```diff
@@ -0,0 +1,585 @@
+                            toImport.add(info.getElements().getTypeElement(LIST_TYPE_NAME));
+                            toImport.add(info.getElements().getTypeElement(MICRONAUT_ARGUMENT_NAME));
+                            toImport.add(tae);
+                            String typeName = typeName(info, type);
+                            String typeArgName = typeName(info, typeArg);
+                            return new String[] {"retrieve", typeName, "list", ",Argument.listOf(" + typeArgName + ".class)"};
+                        }
+                    }
+                    break;
+                case OPTIONAL_TYPE_NAME:
+                    typeArg = declType.getTypeArguments().get(0);
+                    if (typeArg.getKind() == TypeKind.DECLARED) {
+                        TypeElement tae = (TypeElement) ((DeclaredType) typeArg).asElement();
+                        if (Utils.getAnnotation(tae.getAnnotationMirrors(), MICRONAUT_SERDEABLE_ANNOTATION_NAME) != null) {
+                            toImport.add(info.getElements().getTypeElement(OPTIONAL_TYPE_NAME));
+                            toImport.add(info.getElements().getTypeElement(MICRONAUT_ARGUMENT_NAME));
+                            toImport.add(tae);
+                            String typeName = typeName(info, type);
+                            String typeArgName = typeName(info, typeArg);
+                            return new String[] {"retrieve", typeName, varName(typeArgName, false), ",Argument.optionalOf(" + typeArgName + ".class)"};
+                        }
+                    }
+                    break;
+                default:
+                    if (Utils.getAnnotation(te.getAnnotationMirrors(), MICRONAUT_SERDEABLE_ANNOTATION_NAME) != null) {
+                        toImport.add(te);
+                        String typeName = typeName(info, type);
+                        return new String[] {"retrieve", typeName, varName(typeName, false), "," + typeName + ".class"};
+                    }
+            }
+        }
+        toImport.add(info.getElements().getTypeElement(MICRONAUT_HTTP_RESPONSE_NAME));
+        return new String[] {"exchange", "HttpResponse<?>", "response", EMPTY};
+    }
+
+    private static String typeName(CompilationInfo info, TypeMirror tm) {
+        if (tm.getKind() == TypeKind.TYPEVAR) {
+            tm = ((TypeVariable) tm).getUpperBound();
+        }
+        return Utils.getTypeName(info, tm, false, false).toString();
+    }
+
+    private static String varName(String varTypeName, boolean upperCase) {
+        StringBuilder sb = new StringBuilder(varTypeName);
+        char firstChar = sb.charAt(0);
+        sb.setCharAt(0, upperCase ? Character.toUpperCase(firstChar) : Character.toLowerCase(firstChar));
+        return sb.toString();
+    }
+
+    private static String defaultValue(TypeMirror tm) {
+        if (tm != null) {
+            switch(tm.getKind()) {
```

Comment:
Purely for consideration - there is `TypeKind.isPrimitive`, which could be used to avoid the switch. Not a big deal, though.